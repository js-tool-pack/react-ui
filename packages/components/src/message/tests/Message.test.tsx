import Demo from '../demo/useMessageHolder';
import { render, fireEvent } from '@testing-library/react';

describe('Message', () => {
  test('holder', async () => {
    const { container } = render(<Demo />);
    const getRoot = () => document.body.querySelector('.t-transition-group');
    expect(getRoot()).toMatchSnapshot();
    fireEvent.click(container.querySelector('.t-button')!);
    expect(getRoot()).toMatchSnapshot();
  });
});
