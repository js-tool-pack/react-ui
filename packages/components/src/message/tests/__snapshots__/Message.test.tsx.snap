// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Message holder 1`] = `
<ul
  class="t-transition-group t-message-queue"
/>
`;

exports[`Message holder 2`] = `
<ul
  class="t-transition-group t-message-queue"
>
  <li
    class="t-message-queue-enter-active t-message-queue-enter-to"
    style=""
  >
    <div
      class="t-message t-message--info"
    >
      <div
        class="t-message__icon-wrapper"
      >
        <i
          class="t-icon"
          role="img"
        >
          <svg
            viewBox="0 0 1024 1024"
          >
            <path
              d="M512 2.27555555a509.72444445 509.72444445 0 1 0 0 1019.4488889A509.72444445 509.72444445 0 0 0 512 2.27555555z m36.40888889 755.48444445a9.10222222 9.10222222 0 0 1-9.10222222 9.10222222h-54.61333334a9.10222222 9.10222222 0 0 1-9.10222222-9.10222222v-309.47555555c0-5.02442667 4.07779555-9.10222222 9.10222222-9.10222223h54.61333334c5.02442667 0 9.10222222 4.07779555 9.10222222 9.10222223v309.47555555zM512 366.36444445a54.61333333 54.61333333 0 1 1 0-109.22666667A54.61333333 54.61333333 0 0 1 512 366.36444445z"
              fill="currentColor"
            />
          </svg>
        </i>
      </div>
      hello world
    </div>
  </li>
</ul>
`;
